src/ai/backend/storage/filebrowser.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+import json
+import logging
+import pathlib
+from pathlib import Path
+from typing import AsyncIterator
+
+import aiodocker
+import aiofiles
+
+from ai.backend.common.logging import BraceStyleAdapter
+
+from .context import Context
+
+log = BraceStyleAdapter(logging.getLogger(__name__))
+
+__all__ = (
+    "create_or_update",
+    "destroy",
+    "cleanup",
+)
+
+
+def mangle_path(mount_path, vfid):
+    prefix1 = vfid[0:2]
+    prefix2 = vfid[2:4]
+    rest = vfid[4:]
+    return Path(mount_path, prefix1, prefix2, rest)
+
+
+async def create_or_update(ctx: Context, vfolders: list[str]) -> tuple[str, int, str]:
+
+    image = ctx.local_config["filebrowser"]["image"]
+    service_ip = ctx.local_config["filebrowser"]["service-ip"]
+    service_port = ctx.local_config["filebrowser"]["service_port"]
+    settings_path = ctx.local_config["filebrowser"]["settings_path"]
+    mount_path = ctx.local_config["filebrowser"]["mount_path"]
+    cpu_count = ctx.local_config["filebrowser"]["max-cpu"]
+    memory = ctx.local_config["filebrowser"]["max-mem"]
```

Comment:
Please implement `max-containers` configuration.
This would require a local tracking of container states, like the agent, so that we could count the existing number of containers and get notified when the containers terminate by themselves.

Ideally, we could extend a `ai.backend.common.docker` module to provide Docker lifecycle events subscription and basic parsing into typed Python objects, and reuse it across the agent and storage-proxy.